### PR TITLE
feat: signin and signup page 

### PR DESCRIPTION
**Pull request by haritianaadriano**

### CODE REVIEW

**Review by YumeT023 — CHANGES_REQUESTED** [review]
format your code: `yarn format`

File: src/features/auth/schema.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-export type BasicPayload = z.infer<typeof basic>;
+export const user = z.object({
+  email: z.string().email(),
+  username: z.string(),
+  firstname: z.string(),
+  lastname: z.string(),
+  bio: z.string() 
+})
+
+export type BasicPayload = z.infer<typeof login>;
```
Comment (YumeT023) [inline src/features/auth/schema.ts]: remove unused

make a type for every schema you write here:

```js
export type LoginUser = z.infer<typeof loginUser>;
```

File: src/features/auth/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export * from "./store";
-export * from "./components";
+export * from "./components/signin";
```
Comment (YumeT023) [inline src/features/auth/index.ts]: why did you move these components ?

File: src/features/auth/schema.ts [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
   password: z.string().min(8),
 });
 
-export type BasicPayload = z.infer<typeof basic>;
+export const user = z.object({
+  email: z.string().email(),
+  username: z.string(),
+  firstname: z.string(),
+  lastname: z.string(),
+  bio: z.string()
```
Comment (YumeT023) [inline src/features/auth/schema.ts]: bio should be `.optional()`-ed, same for `username`,
also include the ctx where you use it in the variable name, like: `loginUser`, `signupUser`, `updateUser`, ...

File: src/features/auth/components/signin/Login.tsx [hunk 4]
```diff
@@ -45,12 +48,45 @@ export const Login: FC = () => {
...
+    },
+  })
+
   // TODO: build the ui
   // register email and password input
   return (
-    <form onSubmit={handleSubmit(onEmailAndPassword)}>
-      {/* TODO: form inputs */}
+    <Form {...form}>
+      <form onSubmit={handleSubmit(onEmailAndPassword)}>
```
Comment (YumeT023) [inline src/features/auth/components/signin/Login.tsx]: weird... why is there a form inside the `Form` ?

File: src/features/auth/components/signin/Login.tsx [hunk 5]
```diff
@@ -2,18 +2,21 @@ import {FC} from "react";
...
+import { Form, FormControl, FormField, FormItem, FormLabel } from "@/components/shadcn-ui/form";
+import { z } from "zod";
+import { Input } from "@/components/shadcn-ui/input";
 
 export const Login: FC = () => {
   const store = useAuthStore();
   const navigate = useNavigate();
   // TODO: use register with inputs
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const {register: _register, handleSubmit} = useForm<BasicPayload>({
```
Comment (YumeT023) [inline src/features/auth/components/signin/Login.tsx]: remove unused!

File: src/features/auth/components/signin/Login.tsx [hunk 6]
```diff
@@ -45,12 +48,45 @@ export const Login: FC = () => {
     };
   };
 
+  const form = useForm<z.infer<typeof login>>({
```
Comment (YumeT023) [inline src/features/auth/components/signin/Login.tsx]: do type inference inside schema.ts file, use the type here, as the above one
`useForm<Type>(...)`

File: src/features/auth/components/signin/Login.tsx [hunk 7]
```diff
@@ -2,18 +2,21 @@ import {FC} from "react";
 import {SubmitHandler, useForm} from "react-hook-form";
 import {zodResolver} from "@hookform/resolvers/zod";
 import {useAuthStore} from "@/features/auth";
-import {basic, BasicPayload} from "@/features/auth/schema.ts";
```
Comment (YumeT023) [inline src/features/auth/components/signin/Login.tsx]: order your imports

- named_import_from_libs
- named_import_from_local_modules (group them)
- default_imports_from_libs
- default_imports_from_local_modules (group them)

File: src/features/auth/components/signin/Login.tsx [hunk 8]
```diff
@@ -45,12 +48,45 @@ export const Login: FC = () => {
...
+            </FormControl>
+          </FormItem>
+        )}
+      />
+      <FormField
+        control={form.control}
+        name="password"
+        render={({field}) => (
+          <FormItem>
+            <FormLabel>email</FormLabel>
```
Comment (YumeT023) [inline src/features/auth/components/signin/Login.tsx]: you say this is a password but label it as `email`

**Review by YumeT023 — CHANGES_REQUESTED** [review]

**Comment by YumeT023** [comment]
shadcn-ui cmp generatino: `chore(gen): add .... components`
changes evolving ui display: `feat: ...`
imports order, variable name, ... :  `style: ...`

**Comment by YumeT023** [comment]
Seems like a work in progress

image

File: src/features/auth/index.ts (hunk 2 above)
Comment (haritianaadriano) [reply src/features/auth/index.ts]: or i must make sigin and signup in different component directory ?

**Review by YumeT023 — CHANGES_REQUESTED** [review]
No do not overcomplicate things
/features/auth/components will just contains every components (.jsx)
do not add lot of subdirectory

refactor it:

_features/auth/components_
- _Login.tsx_
- _Authenticated.tsx_
- _Signup.tsx_
- _index.tsx_ 

is enough

File: src/features/auth/index.ts (hunk 2 above)
Comment (YumeT023) [reply src/features/auth/index.ts]: You should have them in the same directory,
the `components` directory of the `auth` feature

**Comment by YumeT023** [comment]
Build is ko:
image
